### ISSUE TITLE
two differemt seeds 

### ISSUE THREAD

**Issue by KarolyneFarfan** [Finspire13/DiffAct#6]
Hello,

Thank you for sharing your code, it has been very useful, but currently I am having trouble to understand why you are using 2 different seeds in the function ddim_sample, first in https://github.com/Finspire13/DiffAct/blob/a223e0c27a6f02adfb321a518cb0f7dd0d0896b6/model.py#L323 and the in https://github.com/Finspire13/DiffAct/blob/a223e0c27a6f02adfb321a518cb0f7dd0d0896b6/model.py#L369C1-L369C9

Thank you in advance,
Summy

**Comment by Finspire13** [comment]
Hi Summy,

The first seed is the actual seed to remove randomness for inference. This is fixed.

The second seed is a 'random' random seed to restore the randomness outside this inference function (ddim_sample), since randomness is crucial and we want to keep it for **training**. The code you referred to is a workaround to achieve this 'random' random seed using the current time in milliseconds. This is different every time this function is called.

In short, this implementation is to restrict determinism within this ddim_sample function.
